testing/tests/DevExpress.ui.widgets/treeViewParts/virtualMode.js

```diff
@@ -199,6 +199,39 @@ QUnit.test('Remove toggle icon after expand childless item', function(assert) {
     assert.equal($icons.length, 2);
 });
 
+QUnit.test('No custom expander icons should be visible after expand childless item', function(assert) {
+    const treeView = new TreeView(this.$element, {
+        dataSource: $.extend(true, [], data2),
+        dataStructure: 'plain',
+        expandButtonIcon: 'add',
+        collapseButtonIcon: 'minus',
+        virtualModeEnabled: true
+    });
+
+    const $expandIcons = this.$element.find(`.${internals.CUSTOM_EXPAND_ICON_CLASS}`);
+    assert.ok($expandIcons.eq(0).is(':visible'));
+    assert.ok($expandIcons.eq(1).is(':visible'));
+    assert.ok($expandIcons.eq(2).is(':visible'));
+
+    const $collapseIcons = this.$element.find(`.${internals.CUSTOM_COLLAPSE_ICON_CLASS}`);
+    assert.notOk($collapseIcons.eq(0).is(':visible'));
+    assert.notOk($collapseIcons.eq(1).is(':visible'));
+    assert.notOk($collapseIcons.eq(2).is(':visible'));
+
+    treeView.expandItem(16);
+
+    const $expandIconsAfterExpandChildlessNode = this.$element.find(`.${internals.CUSTOM_EXPAND_ICON_CLASS}`);
+
+    assert.ok($expandIconsAfterExpandChildlessNode.eq(0).is(':visible'));
+    assert.ok($expandIconsAfterExpandChildlessNode.eq(1).is(':visible'));
+    assert.notOk($expandIconsAfterExpandChildlessNode.eq(2).is(':visible'));
+
+    const $collapseIconsAfterExpandChildlessNode = this.$element.find(`.${internals.CUSTOM_COLLAPSE_ICON_CLASS}`);
+    assert.notOk($collapseIconsAfterExpandChildlessNode.eq(0).is(':visible'));
+    assert.notOk($collapseIconsAfterExpandChildlessNode.eq(1).is(':visible'));
+    assert.notOk($collapseIconsAfterExpandChildlessNode.eq(2).is(':visible'));
+});
+
 QUnit.test('Remove loadindicator after expand childless item', function(assert) {
     new TreeView(this.$element, {
```

Comment:
I also suggest u to add tests in `treeView.markup.tests` file to check that necessary icons are rendered if an appropriate property is passed in initial config or property value is changed in runtime, check that all necessary classes are added / toggles and so on